torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -1,14 +1,36 @@
 #include <torch/csrc/distributed/rpc/rpc_agent.h>
+#include <torch/csrc/distributed/autograd/context/dist_autograd_container.h>
+#include <torch/csrc/distributed/autograd/utils.h>
 
 namespace torch {
 namespace distributed {
 namespace rpc {
 
+using namespace torch::distributed::autograd;
+
 RpcAgent::RpcAgent(std::string workerName, RequestCallback cb)
     : workerName_(std::move(workerName)), cb_(std::move(cb)) {}
 
 RpcAgent::~RpcAgent() = default;
 
+std::shared_ptr<FutureMessage> RpcAgent::send(
+    const std::string& to,
+    Message&& message) {
+  // Record appropriate autograd information before sending the message over the
+  // wire.
+  auto& autogradContainer = DistAutogradContainer::getInstance();
+  if (autogradContainer.hasValidContext()) {
+    // Attach the appropriate autograd edges to the tensors found in the
+    // message.
+    auto grad_fn = addSendRpcBackward(message.tensors());
```

Comment:
I am not against adding it here in this PR. But in the future, we might want to decouple `RpcAgent` interface with autograd code. @satgera mentioned that eventually, we might want to have a separate rpc lib so that `RpcAgent` implementations can have a minimum set of dependencies.

Reply:
Yes, I have re-worked this a bit in my follow up PR for the full forward pass.